briefcase/app.py

```diff
@@ -157,11 +157,16 @@ def find_support_pkg(self):
                             self.platform
                         )):
             for item in page.get('Contents', []):
-                candidates.append(item['Key'])
-        try:
-            return S3_URL + sorted(candidates, reverse=True)[0]
-        except IndexError:
-            return None
+                build_number = int(
+                    item['Key'].rstrip('.tar.gz').split('.')[-1].lstrip('b'))
+                if build_number > top_build_number:
+                    top_build_number, top_build = build_number, item['Key']
+        if top_build:
+            try:
+                return S3_URL + top_build
+            except IndexError:
```

Comment:
Under what conditions will this raise an IndexError?

Reply:
This exception handling was already in there, returning `None` just as it still does now (eventually); I was just preserving the existing behavior. `IndexError` results when the requested S3 path does not exist; I suppose there could be an edge case that this catches where a build is released to S3 but then deleted in the midst of this running.